src/hivpy/sexual_behaviour.py

```diff
@@ -130,3 +144,143 @@ def update_sex_groups(self, population: pd.DataFrame):
                         jump_to_new_group = pd.Series(
                             (rands >= Pmin) & (rands < Pmax), index=rred.index)
                         population.loc[index & jump_to_new_group, "sex_behaviour"] = new_group
+
+    # risk reduction factors
+    def init_risk_factors(self, pop_data):
+        n_pop = len(pop_data)
+        self.init_rred_personal(pop_data, n_pop)
+        pop_data["rred_age"] = np.ones(n_pop)  # Placeholder to be changed each time step
+        self.update_rred_age(pop_data)
+        pop_data["rred"] = (self.new_partner_factor *
+                            pop_data["rred_personal"])
+        self.init_rred_adc(pop_data)
+        self.init_rred_diagnosis(pop_data)
+        self.init_rred_population(pop_data)
+        self.init_rred_art_adherence(pop_data)
+        self.init_rred_balance(pop_data)
+
+    def init_rred_art_adherence(self, pop_data):
+        pop_data["rred_art_adherence"] = 1
+
+    def update_rred_art_adherence(self, pop_data):
+        if("art_adherence" in pop_data.columns):
```

Comment:
Is there any scenario that the column doesn't exist? (same for `partnered` below) Or is this because we have not implemented it yet? (in which case let's mark with a FIXME or note it elsewhere; or just add a column with trivial values!)

Reply:
As far as I know this would exist, but we haven't implemented anything around this area yet so I can't guarantee either way. I didn't want to add a trivial column because I didn't want it to be forgotten about later and then have this column being initialised in two places, but I will add a FIXME here that this may be removed once art_adherence has been properly implemented.